cpp/src/cider/exec/nextgen/operators/RowToColumnNode.cpp

```diff
@@ -329,21 +331,38 @@ class ColumnWriter {
 
   JITValuePointer setFixSizeRawData(utils::FixSizeJITExprValue& fixsize_val) {
     if (expr_->get_type_info().get_type() == kBOOLEAN) {
-      auto raw_data_buffer = context_.getJITFunction()->createLocalJITValue(
-          [this]() { return allocateBitwiseBuffer(1); });
+      auto raw_data_buffer = context_.getJITFunction()->createLocalJITValue([this]() {
+        if (!bitwise_bool_) {
+          // Memory padding for AVX2
+          JITValuePointer prev_len(arrow_array_len_);
+          auto padded_len = (prev_len + 31) / 32 * 32;
+          arrow_array_len_.replace(padded_len);
+          auto mem = allocateBitwiseBuffer(1);
```

Comment:
`allocateBitwiseBuffer` add length argument.

Reply:
Some interfaces of the writer and reader need to be decoupled.